packit_service/worker/checker/koji.py

```diff
@@ -0,0 +1,65 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+
+from packit_service.worker.checker.abstract import Checker
+from packit_service.worker.events.enums import GitlabEventAction
+from packit_service.worker.events import (
+    MergeRequestGitlabEvent,
+    PullRequestGithubEvent,
+    PushGitHubEvent,
+    PushGitlabEvent,
+    PushPagureEvent,
+)
+from packit_service.constants import (
+    KOJI_PRODUCTION_BUILDS_ISSUE,
+    PERMISSIONS_ERROR_WRITE_OR_ADMIN,
+)
+from packit_service.worker.reporting import BaseCommitStatus
+from packit_service.worker.handlers.mixin import GetKojiBuildJobHelperMixin
+
+logger = logging.getLogger(__name__)
+
+
+class PermissionOnKoji(Checker, GetKojiBuildJobHelperMixin):
```

Comment:
I would say we check multiple things here: closed merge requests, whether we react to configured branch, the user who triggered the build and whether scratch build is configured: does it make sense to split this or at least rename since this is not only about permissions?

Reply:
Isn't this comment still valid (I see it marked as resolved)?